python/tvm/tir/buffer.py

```diff
@@ -247,7 +247,9 @@ def decl_buffer(
         shape_dtype = shape[0].dtype if hasattr(shape[0], "dtype") else "int32"
         elem_offset = Var("%s_elem_offset" % name, shape_dtype)
     if data is None:
-        data = Var(name, PointerType(PrimType(dtype)), span)
+        # store bool as i8
+        storage_dtype = "int8" if dtype == "bool" else dtype
```

Comment:
Why is there a special case for bool here?

Reply:
I think bool is usually treated as "uint1" though...

Reply:
In the current convention, the bool are stored as int8, so the pointer type of the bool is `i8*`, while bool is represented as `i1` in the IR.

Reply:
Could we just have the `PrimType` constructor do the conversion?

Reply:
This would complicates the logic of the PrimType. Constructors are expected to be simple and "no surprise". Notably we still want to be able to represent `i1` types in the type system. It is just in common platforms i8* is used to store values of i1. 

Considering the above reasons I don't think it is a good idea to have PrimType constructor to do the conversation.